### ISSUE TITLE
[CI] Deprecation warnings cause failures for 2 tests in MixedClusterClientYamlTestSuiteIT in 6.8

### ISSUE THREAD

**Issue by dimitris-athanasiou** [elastic/elasticsearch#42796]
Build link: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+bwc-tests/112/console

Two tests failed:

Reproduce 1:

```
./gradlew :qa:mixed-cluster:v5.6.0#mixedClusterTestRunner -Dtests.seed=50C77143AD9C6ABC -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT -Dtests.method="test {p0=update/11_shard_header/Update check shard header}" -Dtests.security.manager=true -Dtests.locale=es-CL -Dtests.timezone=Asia/Atyrau -Dcompiler.java=12 -Druntime.java=8
```

Failure 1:

```
FAILURE 0.30s | MixedClusterClientYamlTestSuiteIT.test {p0=update/11_shard_header/Update check shard header} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Failure at [update/11_shard_header:23]: got unexpected warning header [
   > 	299 Elasticsearch-6.8.1-SNAPSHOT-8c087df "Usage of internal versioning for optimistic concurrency control is deprecated and will be removed. Please use the `if_seq_no` and `if_primary_term` parameters instead. (request for index [foobar], type [baz], id [1])"
   > ]
   > 	at __randomizedtesting.SeedInfo.seed([50C77143AD9C6ABC:D8934E9903600744]:0)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:412)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:389)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.lang.AssertionError: got unexpected warning header [
   > 	299 Elasticsearch-6.8.1-SNAPSHOT-8c087df "Usage of internal versioning for optimistic concurrency control is deprecated and will be removed. Please use the `if_seq_no` and `if_primary_term` parameters instead. (request for index [foobar], type [baz], id [1])"
   > ]
   > 	at org.elasticsearch.test.rest.yaml.section.DoSection.checkWarningHeaders(DoSection.java:330)
   > 	at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:258)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:405)
   > 	... 37 more
```

Reproduce 2:

```
REPRODUCE WITH: ./gradlew :qa:mixed-cluster:v5.6.0#mixedClusterTestRunner -Dtests.seed=50C77143AD9C6ABC -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT -Dtests.method="test {p0=update/60_refresh/refresh=wait_for waits until changes are visible in search}" -Dtests.security.manager=true -Dtests.locale=es-CL -Dtests.timezone=Asia/Atyrau -Dcompiler.java=12 -Druntime.java=8
```

Failure 2:

```
FAILURE 1.16s | MixedClusterClientYamlTestSuiteIT.test {p0=update/60_refresh/refresh=wait_for waits until changes are visible in search} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Failure at [update/60_refresh:99]: got unexpected warning header [
   > 	299 Elasticsearch-6.8.1-SNAPSHOT-8c087df "Usage of internal versioning for optimistic concurrency control is deprecated and will be removed. Please use the `if_seq_no` and `if_primary_term` parameters instead. (request for index [update_60_refresh_1], type [test], id [update_60_refresh_id1])"
   > ]
   > 	at __randomizedtesting.SeedInfo.seed([50C77143AD9C6ABC:D8934E9903600744]:0)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:412)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:389)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.lang.AssertionError: got unexpected warning header [
   > 	299 Elasticsearch-6.8.1-SNAPSHOT-8c087df "Usage of internal versioning for optimistic concurrency control is deprecated and will be removed. Please use the `if_seq_no` and `if_primary_term` parameters instead. (request for index [update_60_refresh_1], type [test], id [update_60_refresh_id1])"
   > ]
   > 	at org.elasticsearch.test.rest.yaml.section.DoSection.checkWarningHeaders(DoSection.java:330)
   > 	at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:258)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:405)
   > 	... 37 more
```

I could not reproduce the first but I do reproduce the second reliably on my machine.

**Comment by ywelsch** [comment]
@dnhatn is working on a fix here: https://github.com/elastic/elasticsearch/pull/42641

**Comment by dimitris-athanasiou** [comment]
These also failed in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+default-distro-bwc-tests/98/console with the same error:

```
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=update/10_doc/Partial document}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=update/12_result/Update result field}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=update/20_doc_upsert/Doc upsert}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=update/22_doc_as_upsert/Doc as upsert}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=update/40_routing/Routing}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=update/60_refresh/When refresh url parameter is an empty string that means "refresh immediately"}
```

It seems to be the same underlying issue.

**Comment by dnhatn** [comment]
Fixed in #42641
